crates/store/re_log_encoding/src/decoder/mod.rs

```diff
@@ -132,15 +135,39 @@ pub fn read_options(
     Ok((CrateVersion::from_bytes(version), options))
 }
 
+enum Reader<R: std::io::Read> {
+    Raw(R),
+    Buffered(std::io::BufReader<R>),
+}
+
+impl<R: std::io::Read> std::io::Read for Reader<R> {
+    #[inline]
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        match self {
+            Self::Raw(read) => read.read(buf),
+            Self::Buffered(read) => read.read(buf),
+        }
+    }
+}
+
 pub struct Decoder<R: std::io::Read> {
     version: CrateVersion,
     compression: Compression,
-    read: R,
+    read: Reader<R>,
     uncompressed: Vec<u8>, // scratch space
     compressed: Vec<u8>,   // scratch space
 }
 
 impl<R: std::io::Read> Decoder<R> {
+    /// Instantiates a new decoder.
+    ///
+    /// This does not support multiplexed streams.
```

Comment:
We should explain what a multiplexed stream even is. It's a stream containing multiple… recordings? or multiple stores (e.g. a recording and a blueprint)?

Reply:
"Multiplexing" sounds to me that the streams are interleaved. I think "concatenated" is the words we're looking for here.

Reply:
I don't mind "concatenated".